samcli/lib/providers/provider.py

```diff
@@ -977,7 +978,7 @@ def get_function_build_info(
         dockerfile = cast(str, metadata.get("Dockerfile", ""))
         docker_context = cast(str, metadata.get("DockerContext", ""))
         buildable = dockerfile and docker_context
-        loadable = imageuri and Path(imageuri).is_file()
+        loadable = imageuri and check_path_valid_type(imageuri) and Path(imageuri).is_file()
```

Comment:
They should, its because the check_path_valid_type is used in 2 locations, however in each function, they use either `os.path` or`pathlib.Path` to resolve the path, so I have to choose one or the other library to use.

Reply:
Sorry, just to be more clear, why not just use one or the other? if `check_path_valid_type(imageuri)` and `Path(imageuri).is_file()` always return the same thing, we can just pick one instead of having both and `and`ing the result.

If its the same return result, we could just drop the `and Path(imageuri).is_file()`?

Reply:
Same comment as Lucas, overall PR LGTM.

Reply:
In the second case in sam_function_provider, we have to normalize the resource path before we check the file path exists, so we can't directly just check in the `Path(imageuri).is_file()` in the `check_path_valid_type(imageuri)`, because the `check_path_valid_type(imageuri)` purpose is not to check if its a correct file yet, because even if it is false we will return true, its to make sure that the code does not panic when we go in the future to check if it is a correct file, its simply checking the type

Reply:
Updated to be more intuitive on what the function is doing